### ISSUE TITLE
Host checked too often

### ISSUE THREAD

**Issue by sshmanko** [NagiosEnterprises/nagioscore#336]
After Nagios 4.1.1 was stopped for 5 minutes and started again host checks are rescheduled with 10 second interval:

    Mar  2 00:22:22  nagios: Nagios 4.1.1 starting... (PID=9576)
    Mar  2 00:22:36  nagios: HOST ALERT: host-1;DOWN;SOFT;1;PING CRITICAL - Packet loss = 100%
    Mar  2 00:22:46  nagios: HOST ALERT: host-1;DOWN;SOFT;2;PING CRITICAL - Packet loss = 100%
    Mar  2 00:22:53  nagios: HOST ALERT: host-1;DOWN;HARD;3;PING CRITICAL - Packet loss = 100%
    Mar  2 00:22:53  nagios: HOST NOTIFICATION: email;host-1;DOWN;notify-host-by-email;PING CRITICAL - Packet loss = 100%

So transition SOFT1 -> SOFT2 -> HARD3 happened in 17 seconds.

Our expectation is that notification is sent if host is unreachable at least 2 minutes.

Using host template:
 
    define host {
    name                            linux-server    ; The name of this host template
    notifications_enabled           1                   ; Host notifications are enabled
    event_handler_enabled           1                   ; Host event handler is enabled
    event_handler                   host-handler
    flap_detection_enabled          0                   ; Flap detection is enabled
    process_perf_data               1                   ; Process performance data
    retain_status_information       1                   ; Retain status information across program restarts
    retain_nonstatus_information    1                   ; Retain non-status information across program restarts
    check_period                    24x7                ; By default, Linux hosts are checked round the clock
    check_interval                  1                   ; Actively check the host every minute
    retry_interval                  1                   ; Schedule host check retries at 1 minute intervals
    max_check_attempts              3                   ; Check each Linux host 3 times (max)
    check_command                   check-host-alive    ; Default command to check Linux hosts
    notification_period             24x7                ; Send notifications round the clock
    notification_interval           10                  ; Resend notifications every 10 minutes
    notification_options            d,u,r               ; Only send notifications for specific host states
    register                        0                   
    }

    define command{
    command_name    check-host-alive
    command_line    /usr/local/nagios/libexec/check_ping -H $HOSTADDRESS$ -w 500.0,60% -c 1000.0,100% -p 3 -t 6
    }

nagios.cfg
 
    interval_length=60
    use_aggressive_host_checking=0
    use_retained_program_state=1
    use_retained_scheduling_info=1
    service_inter_check_delay_method=s
    max_service_check_spread=5
    service_interleave_factor=s
    host_inter_check_delay_method=s
    max_host_check_spread=5
    max_concurrent_checks=0
    check_result_reaper_frequency=10
    max_check_result_reaper_time=30

**Comment by jfrickson** [comment]
Looking into it.

**Comment by pm426** [comment]
It was not mentioned, but are there any service checks associated with the host that changed state?  This would trigger on-demand host check execution as expected.

https://assets.nagios.com/downloads/nagioscore/docs/nagioscore/4/en/hostchecks.html

**Comment by sshmanko** [comment]
Yes, there are 4 services with 1 minute check intervals. 
So each of these checks forced host check to be run, independent of check_interval of the host object?

**Comment by jfrickson** [comment]
Failed service check forces host check. Closing
